apps/dashboard/src/components/workflow-editor/steps/digest/scheduled-digest.tsx

```diff
@@ -0,0 +1,168 @@
+import { useMemo } from 'react';
+import cronParser from 'cron-parser';
+
+import {
+  getCronBasedOnPeriod,
+  getPeriodFromCronParts,
+  parseCronString,
+  PeriodValues,
+  toCronFields,
+  toUiFields,
+  UiCronFields,
+} from '@/components/workflow-editor/steps/digest/utils';
+import { Period } from '@/components/workflow-editor/steps/digest/period';
+import { NumbersPicker } from '@/components/workflow-editor/steps/digest/numbers-picker';
+import { DaysOfWeek } from '@/components/workflow-editor/steps/digest/days-of-week';
+import { MultiSelect } from '@/components/primitives/multi-select';
+
+const MONTHS_OPTIONS = [
+  { value: 1, label: 'January' },
+  { value: 2, label: 'February' },
+  { value: 3, label: 'March' },
+  { value: 4, label: 'April' },
+  { value: 5, label: 'May' },
+  { value: 6, label: 'June' },
+  { value: 7, label: 'July' },
+  { value: 8, label: 'August' },
+  { value: 9, label: 'September' },
+  { value: 10, label: 'October' },
+  { value: 11, label: 'November' },
+  { value: 12, label: 'December' },
+];
+
+export const ScheduledDigest = ({
+  value,
+  isDisabled,
+  onValueChange,
+  onError,
+}: {
+  value: string;
+  isDisabled?: boolean;
+  onValueChange: (cron: string) => void;
+  onError?: (error: unknown) => void;
+}) => {
+  const period = useMemo(() => {
+    try {
+      const cronParts = parseCronString(value);
+      return getPeriodFromCronParts(cronParts);
+    } catch (e) {
+      onError?.(e);
+      return PeriodValues.MINUTE;
+    }
+  }, [value, onError]);
+
+  const { second, month, dayOfMonth, dayOfWeek, hour, minute } = useMemo(() => {
+    try {
+      const expression = cronParser.parseExpression(value);
+      return toUiFields(expression.fields);
+    } catch (e) {
+      onError?.(e);
+
+      return {
+        second: [],
+        minute: [],
+        hour: [],
+        dayOfMonth: [],
+        month: [],
+        dayOfWeek: [],
+      };
+    }
+  }, [value, onError]);
+
+  const handleValueChange = (fields: Partial<UiCronFields>) => {
+    const cronFields = toCronFields({
+      second,
+      minute,
+      hour,
+      dayOfWeek,
+      dayOfMonth,
+      month,
+      ...fields,
+    });
+
+    onValueChange(cronParser.fieldsToExpression(cronFields).stringify());
+  };
+
+  const handlePeriodChange = (period: string) => {
+    onValueChange(getCronBasedOnPeriod(period as PeriodValues, { second, minute, hour, dayOfWeek, dayOfMonth, month }));
```

Comment:
When the period value changes, we need to generate a new `cron` expression but preserve the previously selected user values.